### PR TITLE
PLATUI-2888 document patch release process

### PR DESCRIPTION
**Pull request by ellamdav**
Also split out MAINTAINERS info from general CONTRIBUTING guide

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+Occasionally, we need to release a patch to earlier major versions of the library, to fix critical issues
+or make mandatory changes for services on earlier versions that can't easily upgrade to the latest.
+An example of this was the mandated switch to use the Tudor Crown branding following the accession of King Charles III.
+
+We use the following workflow for patch releases:
+1. Create a release branch for the new version, eg.
+`git checkout -b release/v4.5.1`
+2. Create a feature branch for the required changes, eg.
+`git checkout -b PLATUI-9999-my-critical-changes`
```
Comment (oscarduignan) [inline MAINTAINERS.md]: what about this, adds the tag to create the branches from, and step to push the release branch to github so it's available as a target of when you push your changes

```suggestion
1. Create a release branch for the new version, eg.
`git checkout -b release/v4.5.1 tags/4.5.0`
2. Push release branch to github, eg. `git push -u origin release/v4.5.1`
3. Create a feature branch for the required changes, eg.
`git checkout -b PLATUI-9999-my-critical-changes tags/4.5.0`
```

File: MAINTAINERS.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Documentation for project maintainers (MDTP Platform UI team)
```
Comment (oscarduignan) [inline MAINTAINERS.md]:
```suggestion
# Documentation for project maintainers (MDTP Platform UI and Design Resources teams)
```

File: MAINTAINERS.md (hunk 1 above)
Comment (ellamdav) [reply MAINTAINERS.md]: ah yes, much more explicit ❤️  I think the last checkout would be off the release branch anyway, so is there a need to specify the tag again?

Comment (ellamdav) [reply MAINTAINERS.md]: updated now, kept it all explicit - would be nice to link to the build job in Jenkins, but I guess we shouldn't do this in a public repo?

Comment (oscarduignan) [reply MAINTAINERS.md]: ah you're right! I didn't think of that 🙌

**Review by oscarduignan — DISMISSED** [review]

File: MAINTAINERS.md (hunk 1 above)
Comment (oscarduignan) [reply MAINTAINERS.md]: yeah, I still don't understand all the rules around linking to stuff - so just assume no links 🤷‍♂️

**Review by oscarduignan — APPROVED** [review]

**Pull request merged**